### DISCUSSION TITLE
User Deactivated by Admin Should Not Be Allowed to Activate Again

### DISCUSSION THREAD

**Discussion by EliMariaaaa** [codeigniter4/shield#1097] [Q&A]
Hello! I'd like to not allow the user to login if the admin deactivated his/her account. I use the `$user->deactivate()` code. But when I login the deactivated account, I get an email activation email. How can I differentiate the email activation for newly registered users and the one that the admin deactivated?

**Comment by warcooft** [comment]
> Hello! I'd like to not allow the user to login if the admin deactivated his/her account. I use the `$user->deactivate()` code. But when I login the deactivated account, I get an email activation email. How can I differentiate the email activation for newly registered users and the one that the admin deactivated?

Try using 
```php 
$user->ban();
```

**Comment by datamweb** [comment]
The activation and deactivation logic is related to email confirmation. If you want to prevent the user from entering, you must block the user. See the link below:
https://shield.codeigniter.com/user_management/banning_users
